radis/io/hitran.py

```diff
@@ -1160,7 +1166,49 @@ def download_all_hitran_isotopes(molecule, directory):
                         )
                         os.remove(join(directory, file + ".data"))
                 try:
-                    fetch(file, get_molecule_identifier(molecule), iso, wmin, wmax)
+                      
+                    if(len(extra_params) != 0):
+                        fetch(file, get_molecule_identifier(molecule), iso, wmin, wmax, Parameters=extra_params) #fetching .data and .header file having extra params (comma separated in .data)
+                        
+                        from hapi import getColumn
+                        import numpy.ma as ma
+                        import pandas as pd
+                    
+                    
+                        file_params = {}  
+                        temp_dict = {}
+                        local_df = pd.DataFrame()
+                        global temp_df
+                        
+                        
+                        
+                        for val in extra_params:
+                            a = ma.masked_equal(getColumn(file,val),1)
+                            a = ma.getdata(a)
+                            temp_dict[val] = a.tolist()
+                        
+                        
+                        
+                        file_params[file + ".data"] = temp_dict  # { 'CO_1.data' : {'gamma_CO2':[....] , 'n_CO2':[....]}}
+                        
+                                          
+                        for param,val in file_params[file +".data"].items():  #add extra params columns in global df
+                            local_df[param] = pd.DataFrame(val)
+                    
+                        temp_df = pd.concat([temp_df, local_df])
+                        
+                        ######delete the extra params comma separated from .data file
+                        import fileinput 
+                    
+                        #local_path = join(directory, file + ".data")
+                        with fileinput.FileInput(join(directory,file +".data"), inplace=True) as temp_file:
+                            for line in temp_file:
+                                split_string = line.split(",",1)
+                                line = split_string[0]
+                                print(line)
```

Comment:
what does this block do ?

Reply:
the delete block -
So, whenever call the fetch method with Parameters= ['gamma_CO2','n_CO2'] the `.data` file contains basic attributes along with the extra parameters passed in a comma-separated manner

when simply fetch() is called 
![image](https://user-images.githubusercontent.com/38047163/179351510-40c74dda-6f52-4d07-98e6-822da3a51ca5.png)


**when fetch() with Parameters** we get extra values comma separated 
![image](https://user-images.githubusercontent.com/38047163/179351518-ddfa299c-97c6-4c34-b5b8-07a953dd0369.png)


So, these extra parameters are extracted using getColumn of HAPI and initially stored in a dataframe which is then converted in a .hdf5 file, this `delete block` basically removes the comma-separated values (which are the extra parameters) so that it does not hinder the normal workflow of parsing the basic attributes/columns.
I went ahead and created the .hdf5 file of extra parameters in this manner so that it doesn't hinder the normal processing of code. Otherwise, we would have to define a different way of getting these extra parameters